source/ShipInfoPanel.cpp

```diff
@@ -169,9 +169,42 @@ bool ShipInfoPanel::KeyDown(SDL_Keycode key, Uint16 mod, const Command &command,
 	else if(panelState.CanEdit() && key == 'D')
 	{
 		if(shipIt->get() != player.Flagship())
-			GetUI()->Push(new Dialog(this, &ShipInfoPanel::Disown, "Are you sure you want to disown \""
+		{
+			map<const Outfit*, int> uniqueOutfits;
+			for(const auto &it : shipIt->get()->Outfits())
+				if(it.first->Attributes().Get("unique") > 0)
+					uniqueOutfits[it.first] = it.second;
+			for(const auto &it : shipIt->get()->Cargo().Outfits())
+				if(it.first->Attributes().Get("unique") > 0)
+					uniqueOutfits[it.first] += it.second;
+
+			string message = "Are you sure you want to disown \""
 				+ shipIt->get()->Name()
-				+ "\"? Disowning a ship rather than selling it means you will not get any money for it."));
+				+ "\"? Disowning a ship rather than selling it means you will not get any money for it.";
+			if(!uniqueOutfits.empty())
+			{
+				const int uniquesSize = uniqueOutfits.size();
+				int detailedOutfitList = 20;
+				if(detailedOutfitList > uniquesSize)
+					detailedOutfitList = uniquesSize;
+				else if(detailedOutfitList < uniquesSize)
+					--detailedOutfitList;
```

Comment:
```suggestion
				const int detailedOutfitList = 20 > uniquesSize ? uniquesSize : 20 - (detailedOutfitList < uniquesSize);
```
Dunno, personal taste do it or don't, I don't care

Reply:
I would propose:

```cpp
const int detailedOutfitSize = (uniqueSize > 20 ? 19 : uniqueSize);
```

Reply:
I am confused, is that the same?

Reply:
Feel free to double check :) (also just fyi your suggestion doesn't work because it uses the variable before it is defined)

Reply:
Oh, right 😅
It seems to me that your suggestion works.